### PR TITLE
Add include_samples=True

### PR DESCRIPTION
**Pull request by beccasaurus**
Add support for GAPIC generated code samples.

You can find a detailed description of what `include_samples=True` does here:
 - [What does include_samples=True do?](https://gist.github.com/beccasaurus/8ac942988a8f6021a6bf938eb0b6858b#what-does-include_samplestrue-do)

This [prototype](https://gist.github.com/beccasaurus/8ac942988a8f6021a6bf938eb0b6858b) was used to send these initial PRs demonstrating generated code samples:
 - https://github.com/googleapis/google-cloud-php/pull/1991
 - https://github.com/googleapis/google-cloud-python/pull/8210
 - https://github.com/googleapis/google-cloud-ruby/pull/3431

> ^--- PRs have since been updated to use `include_samples=True` from this branch.

This is a supported feature of [GAPIC](https://github.com/googleapis/gapic-generator).

The official method of testing generated samples is [sample-tester](https://github.com/googleapis/sample-tester).

Library developers define required resources for code samples in [`sample_resources.yaml`](https://github.com/googleapis/googleapis/blob/master/google/cloud/speech/sample_resources.yaml) files.

Support for `include_samples=True` based on implementation of `include_protos=True`.

---

### `TODO(beccasaurus)`

 - [ ] Add `include_samples` to README
 - [ ] Update sample manifest so you can call sample-tester from any directory
    (Requires in-progress implementation of: https://github.com/googleapis/sample-tester/issues/73)
    (Based on feedback [from PHP](https://github.com/googleapis/google-cloud-php/pull/1991))
 - [ ] Show example of working Java (PR to go along with this), ensure the gen-manifest is correct
 - [ ] Demonstrate working with pre-existing hand-written samples `+` generated samples

---

/cc @googleapis/samplegen @dwsupplee 

---

> [go/library-samples](http://go/library-samples) (internal)

### CODE REVIEW

File: synthtool/gcp/gapic_generator.py [hunk 1]
```diff
@@ -193,3 +204,112 @@ def _clone_googleapis_private(self):
...
+
+        # Do not proceed if genfiles does not include samples/{version} dir.
+        if not samples_version_dir.is_dir():
+            return None
+
+        import os
+        import requests
+        import shutil
+        import yaml
+        from synthtool import shell
```
Comment (busunkim96) [inline synthtool/gcp/gapic_generator.py]: Could imports be moved up to the top?

**Comment by beccasaurus** [comment]
Kokoro has been stuck on: _Expected — Waiting for status to be reported_.  
Is there anything I should do to kick it to run? 

Let me know if I can change anything / provide any information to aid in review.

Here are some considerations that I made:
 - Follow the exact same approach as include_protos=True for consistency
 - Everything is put under genfiles.  
    Nothing pollutes the workspace where synth was invoked from!
 - `requests` is used for pulling down files for simplicity and to be idiomatic  
    (I [tried using GCS client](https://github.com/googleapis/synthtool/pull/263/commits/d881f069af2bd328ab873e2ca64cba442e4d7825) and didn't like it)
 - Fails happily if testing tool isn't available (similar to how I've seen blacken fail, for example).
    Doesn't blow up, warning only.

The method is just barely shorter than the `_generate_code` method.  
If it should be extracted to a class, lemme know. Available to make any requested changes!

**Comment by busunkim96** [comment]
@beccasaurus LGTM! I think our linter is unhappy, but a quick `nox -s blacken` should fix that. :)

**Comment by beccasaurus** [comment]
~ _Just pushed a number of changes_ ~

This is **production ready for: 🐍 Python, 🐘 PHP, 💎 Ruby, and 🚀 Node.js**

This is **ready for re-review** 🔍

> Note: Support for ☕️ **Java** and #️⃣ **C#** explicitly de-scoped (will add later)

----

☑️ [**Response to Feedback**]

Since the PR was first sent, there have been a number of changes made  
to the sample generation tooling as a response to user and reviewer feedback:

 - `sample-tester` can be called from any directory
   - This was a response to feedback from the [PHP](https://github.com/googleapis/google-cloud-php/pull/1991) samplegen PR
   - This makes it simpler to run sample tests from CI (e.g. Kokoro, CircleCI, Travis, etc)
   - Before this, sample-tester needed to be run from certain directories. Now it knows the location of sample files relative to the project root and can be invoked from anywhere.

 - `sample-tester` no longer requires passing individual test files
    - you can simply run `$ sample-tester` or `$ sample-tester samples/`
    - this should be more familiar to developers accustomed to other test runners  
      (e.g mocha/ava, pytest, junit, phpunit, rspec/minitest, go test, etc etc)

     ^--- changes relevant to integrating samples into client library repositories

----

ℹ️ [**Misc**]

There have been a number of other, unrelated improvements to the tooling:

 - samples can now be authored in their own, `standalone .yaml files`
   - no longer required to configure samples in _gapic.yaml configs _(which is being deprecated)_
   - samples are authored in a new, improved, easier to read and write .yaml format
     > _note: the samples in googleapis/googleapis do not yet reflect the new format_
 - all YAML files now have a `type:` and schema `version` for tooling production readiness
   - using this for backwards compatible support of old sample/test/manifest formats
 - Java packaging improvements
 - C# is now supported

Java + C# support will be added in the future (out of scope for this PR)

----
/cc @googleapis/samplegen 🦇

**Comment by beccasaurus** [comment]
@busunkim96 Kokoro is happy, but FYI: I tweaked the since the last review, would love another look-see. Nothing more to add on my end, it's in a happy state.

**Comment by beccasaurus** [comment]
Ty kindly!

**Pull request merged**